### PR TITLE
use kubectl from the lower layers in the kubeadm image

### PR DESCRIPTION
**Pull request by BenTheElder**
... also bump the images to go with this and add `--no-install-recommends`

/assign @pipejakob @luxas 

cc @liggitt 

/hold

### CODE REVIEW

File: images/kubeadm/Dockerfile [hunk 1]
```diff
@@ -49,11 +48,6 @@ RUN mkdir -p /tmp/terraform/ && \
     ) && \
     rm -rf /tmp/terraform
 
-# TODO(pipejakob): Instead of fetching a specific precompiled version, get
```
Comment (BenTheElder) [inline images/kubeadm/Dockerfile]: I *think* this is fine because we ship kubectl in the underlying image.

Comment (BenTheElder) [reply images/kubeadm/Dockerfile]: @pipejakob @luxas

File: images/kubeadm/Dockerfile [hunk 2]
```diff
@@ -12,14 +12,14 @@
...
-FROM gcr.io/k8s-testimages/kubekins-e2e:v20180605-65a672b59-master
+FROM gcr.io/k8s-testimages/kubekins-e2e:v20180625-2232199ea-master
 LABEL maintainer "beacham@google.com"
 
 # add env we can debug with the image name:tag
 ARG IMAGE_ARG
 ENV IMAGE=${IMAGE_ARG}
 
-RUN apt-get update && apt-get install -y \
+RUN apt-get update && apt-get install -y --no-install-recommends \
```
Comment (BenTheElder) [inline images/kubeadm/Dockerfile]: we should almost definitely do this to keep the image size down.

**Comment by BenTheElder** [comment]
#8405 :(
/retest

File: images/kubeadm/Dockerfile [hunk 3]
```diff
@@ -30,7 +30,6 @@ RUN apt-get update && apt-get install -y \
 
 ENV TERRAFORM_VERSION=0.9.4 \
     KUBERNETES_PROVIDER=kubernetes-anywhere \
-    KUBECTL_VERSION=1.6.4 \
```
Comment (luxas) [inline images/kubeadm/Dockerfile]: no idea why this was added here but hmm
/shrug

File: prow/config.yaml [hunk 4]
```diff
@@ -429,7 +429,7 @@ presubmits:
       preset-k8s-ssh: "true"
     spec:
       containers:
-      - image: gcr.io/k8s-testimages/kubekins-e2e:v20180605-65a672b59-master
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20180625-2232199ea-master
```
Comment (luxas) [inline prow/config.yaml]: How is bumping kubekins related to this change? This seem to touch the kubeadm image only

File: images/kubeadm/Dockerfile (hunk 1 above)
Comment (luxas) [reply images/kubeadm/Dockerfile]: hmm, this is the kubectl used when `e2e.test` execs out, right?
I'm not sure how `e2e.test` is populated inside of this image and/or if kubectl is copied over from somewhere in that process.
I guess we can try to merge this change and see if stuff breaks or not :wink:

File: prow/config.yaml (hunk 4 above)
Comment (krzyzacy) [reply prow/config.yaml]: kubeadm is always built on top of kubekins, we'll keep both of them up-to-date

File: images/kubeadm/Dockerfile (hunk 1 above)
Comment (BenTheElder) [reply images/kubeadm/Dockerfile]: no, the tests should be using the kubectl from the current build under test, this kubectl shouldn't be used for much of anything (other tests in some other job not using kubetest??)

Comment (luxas) [reply images/kubeadm/Dockerfile]: let's go with this and see what happens!

File: images/kubeadm/Dockerfile (hunk 3 above)
Comment (BenTheElder) [reply images/kubeadm/Dockerfile]: yeah, I mean we could keep upgrading this but... there's also a kubectl provided in the image layer underneath (kubekins-e2e), which currently tacks the stable one gcloud ships but could be switched for any other stable kubectl

**Comment by BenTheElder** [comment]
I'll fix up the image bump later, we should be pretty free to go for it even if we did think it would break since we're just post-release now.

File: prow/config.yaml (hunk 4 above)
Comment (BenTheElder) [reply prow/config.yaml]: we have a convenient script to upgrade the e2e images to the current kubetest source etc, usually if we upgrade e2e images we just do that. we *can* manually just bump the kubeadm image if we want to do that though. realistically right now we should be safe to upgrade everything and we want to do that regularly.

**Comment by BenTheElder** [comment]
rebased and re-bumped images

**Comment by BenTheElder** [comment]
er rebase was not quite up to date ? rebased again and re-bumped.

**Comment by BenTheElder** [comment]
ahhh discussing offline with @krzyzacy, the image bump script doesn't support the jobs split out from the main config yet. need to fix that first.

**Comment by BenTheElder** [comment]
OK, can we get this in now @krzyzacy ? :upside_down_face:

**Comment by BenTheElder** [comment]
gah, opened https://github.com/kubernetes/test-infra/pull/8597

**Comment by BenTheElder** [comment]
fixed the config here, seperately will deal with https://github.com/kubernetes/test-infra/pull/8597/files#r200754404

**Comment by BenTheElder** [comment]
tests are finally green again
/shrug

**Comment by krzyzacy** [comment]
hummmmm this comment indent is wrong again ?!

**Pull request merged**